docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,187 @@
+Workflows were open when you added the first version. If you deployed any Workers with a _different_
+version, those Workers receive no Tasks.
+
+Now we need to change the Workflow for some reason, so we add `2.0` to the sets as the new default:
+
+| set 1   | set 2 (d) |
+| ------- | --------- |
+| 1.0 (d) | 2.0 (d)   |
+
+All new Workflows started on the Task Queue have their first Tasks assigned to version `2.0`.
+Existing `1.0` Workflows keep generating Tasks targeting `1.0`.
+Each deployment of Workers receive their respective Tasks.
+This same concept carries forward for each new incompatible version.
+
+Maybe we have a bug in `2.0`, and we want to make sure all open `2.0` Workflows switch to some new
+code as fast as possible. So we add `2.1` to the sets, marking it as compatible with `2.0`. Now our
+sets look like this:
+
+| set 1   | set 2 (d) |
+| ------- | --------- |
+| 1.0 (d) | 2.0       |
+|         | 2.1 (d)   |
+
+All new Workflow Tasks that are generated for Workflows whose last Workflow Task completion was on
+version `2.0` are now assigned to version `2.1`. Because we specified that `2.1` is compatible with
+`2.0`, Temporal Server assumes that Workers with this version can process the existing Event
+Histories successfully.
+
+We continue with our normal development cycle, adding a `3.0` version.
+Nothing new here:
+
+| set 1   | set 2   | set 3 (d) |
+| ------- | ------- | --------- |
+| 1.0 (d) | 2.0     | 3.0 (d)   |
+|         | 2.1 (d) |           |
+
+Now imagine that version `3.0` doesn't have an explicit bug, but something about the business logic
+is less than ideal. We are OK with existing `3.0` Workflows running to completion, but we want new
+Workflows to use the old `2.x` branch. This operation is supported by performing an update targeting
+`2.1` (or `2.0`) and setting it's set as the current default, which results in these sets:
+
+| set 1   | set 3   | set 2 (d) |
+| ------- | ------- | --------- |
+| 1.0 (d) | 3.0 (d) | 2.0       |
+|         |         | 2.1 (d)   |
+
+Now new workflows start on `2.1`.
+
+#### Permitted and forbidden operations on version sets
+
+A request to change the sets can do one of the following:
+
+- Add a version to the sets as new the default version in a new overall-default compatible set.
+- Add a version to an existing set that is compatible with an existing version.
+  - Optionally making it the default for that set
+  - Optionally making that set the overall-default set.
+- Promote a version within an existing set to become the default for that set.
+- Promote a set to become the overall-default set.
+
+You can't explicitly delete versions. This helps you avoid the situation in which Workflows
+accidentally become stuck with no means of making progress because the version they are associated
+with no longer exists.
+
+However, sometimes you might want to do this intentionally. If you _want_ to make sure that all
```

Comment:
Nice, this is effectively task queue pause, a feature that I've seen requested before.

Reply:
Yes, I've mentioned this one or twice